google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/ConsistencyParams.java

```diff
@@ -0,0 +1,68 @@
+package com.google.cloud.bigtable.admin.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.admin.v2.CheckConsistencyRequest;
+import com.google.bigtable.admin.v2.TableName;
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nonnull;
+
+public class ConsistencyParams {
+    public enum ConsistencyMode {
+        /**
+         * Checks that reads using an app profile with `StandardIsolation` can
+         * see all writes committed before the token was created, even if the
+         * read and write target different clusters.
+         */
+        STANDARD(CheckConsistencyRequest.ModeCase.STANDARD_READ_REMOTE_WRITES),
+
+        /**
+         * Checks that reads using an app profile with `DataBoostIsolationReadOnly`
+         * can see all writes committed before the token was created, but only if
+         * the read and write target the same cluster.
+         */
+        DATA_BOOST(CheckConsistencyRequest.ModeCase.DATA_BOOST_READ_LOCAL_WRITES);
+
+        @Nonnull
+        private final CheckConsistencyRequest.ModeCase proto;
+
+        ConsistencyMode(CheckConsistencyRequest.ModeCase proto) {
+            this.proto = proto;
+        }
+
+        /**
+         * Creates the request protobuf. This method is considered an internal implementation detail and
+         * not meant to be used by applications.
+         */
+        @InternalApi
+        public CheckConsistencyRequest.ModeCase toProto() {
+            return proto;
+        }
+    }
+
+    private final TableName tableName;
+
+    private final ConsistencyMode consistencyMode;
+
+    ConsistencyParams(TableName tableName, ConsistencyMode mode) {
+        Preconditions.checkNotNull(tableName);
+        Preconditions.checkNotNull(mode);
+        this.tableName = tableName;
+        this.consistencyMode = mode;
+    }
+
+    public static ConsistencyParams of(TableName tableName, ConsistencyMode mode) {
```

Comment:
The client is anchored to an instance, so this should be the relative tableId & the toProto method should take the project & instance ids as a param. See here for an example:
https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/RowMutation.java#L251